dspace-spring-rest/src/main/java/org/dspace/app/rest/RelationshipRestController.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.converter.RelationshipConverter;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.RelationshipRest;
+import org.dspace.app.rest.model.RelationshipRestWrapper;
+import org.dspace.app.rest.model.hateoas.RelationshipResourceWrapper;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Item;
+import org.dspace.content.Relationship;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.RelationshipService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.core.Context;
+import org.dspace.util.UUIDUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This will be the entry point for the api/core/relationships endpoint with additional paths to it
+ */
+@RestController
+@RequestMapping("/api/core/relationships")
+public class RelationshipRestController {
+
+    /**
+     * Regular expression in the request mapping to accept a string as identifier but not the other kind of
+     * identifier (digits or uuid)
+     */
+    private static final String REGEX_REQUESTMAPPING_LABEL = "/{label:^(?!^\\d+$)" +
+        "(?!^[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}$)[\\w+\\-]+$+}";
```

Comment:
This looks strange and is currently limiting the name of the relationships.

I have seen, that the RestContract was already adjusted in https://github.com/DSpace/Rest7Contract/pull/40/commits/874ee072eded2656466ef2f3b79fdce971e6ac06, so this should not be an issue, once the implementation updated.

Reply:
This will indeed be replaced with 
https://github.com/DSpace/Rest7Contract/commit/874ee072eded2656466ef2f3b79fdce971e6ac06